calculator.rb

```diff
@@ -0,0 +1,79 @@
+# add numbers 
+def add(first_num, second_num)
+  return (first_num) + (second_num)
+end 
+
+# subtract numbers
+def subtract(first_num, second_num)
+  return (first_num) - (second_num)
+end 
+
+# multiply numbers 
+def multiply(first_num, second_num)
+  return (first_num) * (second_num)
+end 
+
+# divide numbers 
+def divide(first_num, second_num)
+  if second_num == 0
+    puts "You can't divide by zero."
+    exit 
+  else
+    return (first_num / second_num)
+  end
+end 
+
+# checking if the string from the beginning to the end contains only digits one or more times 
+def check_valid_number(user_number)
+  # if user input contains a digit, convert to an integer. otherwise, display error message
+  if /^\d+$/.match(user_number)
+    return user_number.to_i
+  else 
+    puts "That's not a valid number. Try again."
+    exit 
+  end 
+end 
+
+# ask user for math operation
+puts "Welcome to the calculator program!"
+puts "Please enter two numbers to work with."
+
+# ask user first number for math operation
+puts "First number:"
```

Comment:
From a user experience point of view, it feels a little more natural to choose the operation before the numbers, but that could just be a personal preference. Considering adding spaces between lines to make this a little more user friendly.